coredocument/read_acls.go

```diff
@@ -54,3 +61,59 @@ func appendRole(roles []*coredocumentpb.RoleEntry, role *coredocumentpb.Role) []
 		Role:    role,
 	})
 }
+
+// ReadAccessValidator defines validator functions for peer.
+type ReadAccessValidator interface {
+	PeerCanRead(cd *coredocumentpb.CoreDocument, peer identity.CentID) error
+}
+
+// readAccessValidator implements ReadAccessValidator.
+type readAccessValidator struct{}
+
+// PeerCanRead validate if the core document can be read by the peer.
+// Returns an error if not.
+func (r readAccessValidator) PeerCanRead(cd *coredocumentpb.CoreDocument, peer identity.CentID) error {
```

Comment:
I would maybe use `bool` here.

```suggestion
func (r readAccessValidator) PeerCanRead(cd *coredocumentpb.CoreDocument, peer identity.CentID) bool {
```